internal/k8s/deployments.go

```diff
@@ -103,25 +113,61 @@ func NewObjectRef(obj client.Object) *v1.ObjectReference {
 	}
 }
 
+// ComputeVersionID generates a version ID from the worker deployment spec
+func ComputeVersionID(w *temporaliov1alpha1.TemporalWorkerDeployment) string {
+	return ComputeWorkerDeploymentName(w) + versionIDSeparator + ComputeBuildID(w)
+}
+
+func ComputeBuildID(w *temporaliov1alpha1.TemporalWorkerDeployment) string {
+	if containers := w.Spec.Template.Spec.Containers; len(containers) > 0 {
+		if img := containers[0].Image; img != "" {
+			shortHashSuffix := k8sResourceNameSeparator + utils.ComputeHash(&w.Spec.Template, nil, true)
```

Comment:
I think this is slightly easier to read if you build the suffix after you build the imagePrefix.

Reply:
the reason it is first, is because I need the length of the suffix in order to decide how much to truncate the prefix! and I don't want to hardcode the length of the short hash suffix, in case we want to change it later. 

Does that make sense?